fishtest/fishtest/rundb.py

```diff
@@ -312,9 +309,9 @@ def update_task(self, run_id, task_id, stats, nps, spsa):
                                   drawelo=sprt['drawelo'])
       if sprt_stats['finished']:
         run['args']['sprt']['state'] = sprt_stats['state']
-        self.runs.save(run)
+        self.stop_run(run_id, run)
 
-        self.stop_run(run_id)
+    self.runs.save(run)
```

Comment:
We can have a very important optimization here for expensive SPSA tests:

    if not 'spsa' in run['args'] or spsa_games == spsa['num_games']:
      self.runs.save(run)

because we only need to save the data (and it is a LOT of data for a many parameter SPSA)  at the end of an iteration
or if it is a regular SPRT run!

Reply:
How about:
```
if not 'spsa' in run['args'] or spsa_games >= spsa['num_games']:
  self.runs.save(run)
```
Is >= safer or riskier?

Reply:
Ah, I see. It's the same test as higher up. OK, I go with your version.